src/raftstore/store/store.rs

```diff
@@ -1691,17 +1691,14 @@ impl<T: Transport, C: PdClient> Store<T, C> {
     }
 
     fn handle_snap_mgr_gc(&mut self) -> Result<()> {
-        let mut snap_keys = try!(self.snap_mgr.list_snap());
+        let mut snap_keys = try!(self.snap_mgr.list_idle_snap());
```

Comment:
can we meet a case like:

1. list_idle_snap returns an idle snapshot key
2. Other thread then registers the snapshot key

So unluckily, GC will still delete the snapshot here.